### PR TITLE
test: Code Coverage: Team Advances 

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 1c5c43c</samp>

This pull request refactors the team advance page component and its unit test file to improve code quality and test coverage. It removes unused code, adds mock data for a null project name scenario, and focuses on the team advance test suite.

### <samp>🤖 Generated by Copilot at 1c5c43c</samp>

> _Sing, O Muse, of the skillful coder who devised_
> _A mock data object for the null project name case_
> _And tested the team advance page with focused eyes_
> _To filter and display the data with due grace_

### <samp>🤖 Generated by Copilot at 1c5c43c</samp>

*  Remove unused import of `IonRefresherCustomEvent` from `team-advance.page.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2462/files?diff=unified&w=0#diff-37d36790cfe7df23421ca57de6445676f16b8c5bd3d6e08edd56715feff48905L18))
*  Update type of `event` parameter in `changeState` method of `team-advance.page.ts` to be more generic and compatible with mock event object ([link](https://github.com/fylein/fyle-mobile-app/pull/2462/files?diff=unified&w=0#diff-37d36790cfe7df23421ca57de6445676f16b8c5bd3d6e08edd56715feff48905L148-R147))
*  Remove optional chaining operators from `event.target.complete` and `projectField.field_name` expressions in `team-advance.page.ts` as they are not needed ([link](https://github.com/fylein/fyle-mobile-app/pull/2462/files?diff=unified&w=0#diff-37d36790cfe7df23421ca57de6445676f16b8c5bd3d6e08edd56715feff48905L159-R158), [link](https://github.com/fylein/fyle-mobile-app/pull/2462/files?diff=unified&w=0#diff-37d36790cfe7df23421ca57de6445676f16b8c5bd3d6e08edd56715feff48905L166-R165))
*  Add mock data object for testing null project name field in `expense-field.data.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2462/files?diff=unified&w=0#diff-63f3fbfb622689c5541e2857a5aab121d2cc19049d2bdc7d0d0f6238bdff90bdR810-R831))
*  Modify unit test file for `team-advance.page.ts` to import mock data objects and dependencies and focus on test suite ([link](https://github.com/fylein/fyle-mobile-app/pull/2462/files?diff=unified&w=0#diff-e3bdefa6afbffac13011d029d042d2470afe09fde5089e18758eaadfe91cd803L1-R332))

## Clickup
https://app.clickup.com/t/85zu0c8pb

## Code Coverage
`97.72% Statements 86/88`
`96% Branches 24/25`
`96.29% Functions 26/27`
`97.64% Lines 83/85`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/team-advance/team-advance.page.spec.ts [hunk 1]
```diff
@@ -1,26 +1,333 @@
...
+    });
+  });
+
+  it('onTaskClicked(): should take the user to dashboard and show tasks', () => {
+    component.onTaskClicked();
+
+    expect(router.navigate).toHaveBeenCalledOnceWith(['/', 'enterprise', 'my_dashboard'], {
+      queryParams: { state: 'tasks', tasksFilters: 'none' },
+    });
+    expect(trackingService.tasksPageOpened).toHaveBeenCalledOnceWith({
```
Comment (Julias0) [inline src/app/fyle/team-advance/team-advance.page.spec.ts:322]: we dont need tests for trackingService

File: src/app/fyle/team-advance/team-advance.page.spec.ts [hunk 2]
```diff
@@ -1,26 +1,333 @@
...
+
+    it('should generate params for approved state', () => {
+      const result = component.getExtraParams([AdvancesStates.approved]);
+
+      expect(result).toEqual({
+        areq_approval_state: ['ov.{APPROVAL_PENDING,APPROVAL_DONE}'],
+      });
+    });
+
+    it('should generate params for other states', () => {
```
Comment (Julias0) [inline src/app/fyle/team-advance/team-advance.page.spec.ts:296]: mention rejected in the test case

**Review by Julias0 — APPROVED** [review]
